ui/src/api/contracts.ts

```diff
@@ -258,7 +258,7 @@ export async function addValidator(
     percentToValidator: Math.round(Number(values.percentToValidator) * 10000),
     validatorCommissionAddress: values.validatorCommissionAddress,
     minEntryStake: AlgoAmount.Algos(Number(values.minEntryStake)).microAlgos,
-    maxAlgoPerPool: 0n,
+    maxAlgoPerPool: AlgoAmount.Algos(Number(values.maxAlgoPerPool || 0) * 1_000_000).microAlgos,
```

Comment:
I think this should just be:
AlgoAmount.Algos(Number(values.maxAlgoPerPool || 0).microAlgos

Reply:
We need to multiply the amount set in the form (e.g., "50") to millions first, then convert to microalgos. The `values` object contains the form inputs.

Reply:
ah - for some reason it looked like you were multiplying by 1m to get micro algos..  forgot about the 50 -> 50m conversion.